src/Native/System.Native/pal_networking.cpp

```diff
@@ -27,6 +27,11 @@
 #include <unistd.h>
 #include <vector>
 
+#if !HAVE_IN_PKTINFO
+typedef in_addr in_pktinfo;
+#define IP_PKTINFO IP_RECVDSTADDR
+#endif
+
```

Comment:
Since you still end up needing a #if other than this, I don't think this is worth it. If it were enough to get everything else to compile, I'd reconsider.